tools/retry_cmd.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/sh
+#
+# retry_cmd.sh [-n NLOOPS] [-s SLEEP] CMD
+#
+# Retry command until success or pre-specified number of failures
+#
+# 2018 Chris Markiewicz
+# Released into public domain
+
```

Comment:
Doesn't this need to set `set +o pipefail` here to work? After some digging, I think the original problem reported in #2489 is just that the steps are run with `set -eo pipefile` (circle has started to run it with 2.0). Using both -eo will make the script to exit even with that `|| true` pipe.

Reply:
No, because those apply to each instance of the shell. It doesn't infect subprocesses.